okio/src/commonMain/kotlin/okio/FileHandle.kt

```diff
@@ -35,49 +35,102 @@ package okio
 @ExperimentalFileSystem
 abstract class FileHandle : Closeable {
   /**
-   * True once the file handle is closed. Resources should be released with [closeInternal] once
+   * True once the file handle is closed. Resources should be released with [protectedClose] once
    * this is true and [openStreamCount] is 0.
    */
   private var closed = false
 
   /**
    * Reference count of the number of open sources and sinks on this file handle. Resources should
-   * be released with [closeInternal] once this is 0 and [closed] is true.
+   * be released with [protectedClose] once this is 0 and [closed] is true.
    */
   private var openStreamCount = 0
 
   /**
-   * Removes at least 1, and up to [byteCount] bytes from this and appends them to [sink]. Returns
-   * the number of bytes read, or -1 if this file is exhausted.
+   * Reads at least 1, and up to [byteCount] bytes from this starting at [fileOffset] and copies
+   * them to [array] at [arrayOffset]. Returns the number of bytes read, or -1 if [fileOffset]
+   * equals [size].
    */
   @Throws(IOException::class)
-  abstract fun read(offset: Long, sink: Buffer, byteCount: Long): Long
+  fun read(
+    fileOffset: Long,
+    array: ByteArray,
+    arrayOffset: Int,
+    byteCount: Int
+  ): Int {
+    synchronized(this) {
+      check(!closed) { "closed" }
+    }
+    return protectedRead(fileOffset, array, arrayOffset, byteCount)
```

Comment:
Should you check arrayOffset and byteCount are valid for array? Or trust protectedRead to do that?

Reply:
Yeah, I think that’s protectedRead’s job.